### DISCUSSION TITLE
LAB 02.03 TASK 2

### DISCUSSION THREAD

**Discussion by JMARREDONDO** [rramosp/ai4eng.v1#298] [02 - PYTHON]
Hola profe, estaba realizando el ejercicio, la idea es hacer tres listas cada una que contenga la media, el máximo y el mínimo de cada categoría. Lo verifiqué manualmente y el resultado se acomoda a mi código. Pero a la hora de  ponerlo en ejecución para la nota me arroja un error que no logró comprender porque se da.
![image](https://github.com/rramosp/ai4eng.v1/assets/163493911/83302bdb-9fe4-4528-bab5-83979c6a9d9e)

Este es mi código y su respuesta es la siguiente:
![image](https://github.com/rramosp/ai4eng.v1/assets/163493911/7e51594a-1d04-448d-884d-e422a4225c7d)

Y el error que me arroja es el siguiente:
![image](https://github.com/rramosp/ai4eng.v1/assets/163493911/fc0c8f85-f71a-4a22-97eb-b30747479bdb)

Agradezco su ayuda.

**Comment by Felipe-RA** [comment]
Hola @JMARREDONDO 

Gracias por incluir tu codigo, antes de responderte recordarte que 

```python
cuando utilizas los bloques de código es más fácil poder ayudarte! 
```

Volviendo a tu problema, el error es de tipo `NameError`. Este problema suele ocurrir cuando no declaramos una variable, función, o librería.

En tu caso, el problema es que en el *contexto* de tu función no fue definido en ningún momento la variable d (que utilizas a partir de la línea `9`)

De hecho, estos años el manejo de errores en python ha evolucionado y ahora te dice que posiblemente **te referías a df en lugar de d** 

Recuerda que cuando se califican tus funciones, se llevan a un ambiente de ejecución distinto al que tienes en tu notebook, así que si por fuera de la función definiste la variable `d`, el grader no puede accceder a ella.

Eso es todo, hasta pronto!

**Reply by JMARREDONDO** [reply]
Hola Felipe, ya logré detectar el error gracias a tu comentario.
Muchas gracias.
